File: specification/specs/document/multiple/reindex_on_server/ReindexOnServerRequest.ts
@rest_spec_name("reindex")
class ReindexOnServerRequest extends RequestBase {
	conflicts: Conflicts;
	dest: ReindexDestination;
	script: Script;
	size: long;
	max_docs: long;
	source: ReindexSource;
	@request_parameter()
	refresh: boolean;
	@request_parameter()
	requests_per_second: long;
	@request_parameter()
	scroll: Time;
	@request_parameter()
	slices: long;
	@request_parameter()
	timeout: Time;
	@request_parameter()
	wait_for_active_shards: string;
	@request_parameter()
	wait_for_completion: boolean;
}

File: output/csharp/Document/Multiple/ReindexOnServer/ReindexOnServerRequest.cs
using Nest.Internal;
using Nest.CommonOptions;
using Nest.Common;
using Nest.Document;
using Nest.CommonAbstractions;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.Document {

	public class ReindexOnServerRequest : RequestBase {
		
		[DataMember(Name="refresh")]
		public bool Refresh { get; set; }


		[DataMember(Name="requests_per_second")]
		public long RequestsPerSecond { get; set; }


		[DataMember(Name="scroll")]
		public Time Scroll { get; set; }


		[DataMember(Name="slices")]
		public long Slices { get; set; }


		[DataMember(Name="timeout")]
		public Time Timeout { get; set; }


		[DataMember(Name="wait_for_active_shards")]
		public string WaitForActiveShards { get; set; }


		[DataMember(Name="wait_for_completion")]
		public bool WaitForCompletion { get; set; }


		[DataMember(Name="conflicts")]
		public Conflicts Conflicts { get; set; }


		[DataMember(Name="dest")]
		public ReindexDestination Dest { get; set; }


		[DataMember(Name="max_docs")]
		public long MaxDocs { get; set; }


		[DataMember(Name="script")]
		public Script Script { get; set; }


		[DataMember(Name="size")]
		public long Size { get; set; }


		[DataMember(Name="source")]
		public ReindexSource Source { get; set; }

	}
}
